### ISSUE TITLE
Path to config issue

### ISSUE THREAD

**Issue by skennon10** [LLNL/spack#26169]
This value points to the full path of spack.yaml:
https://github.com/spack/spack/blob/76ad001c563bfdd1821b590110cbe83e1b177d03/lib/spack/spack/cmd/config.py#L255
I think it should get the os.path.dirname(scope.path), since 
https://github.com/spack/spack/blob/76ad001c563bfdd1821b590110cbe83e1b177d03/lib/spack/spack/cmd/config.py#L240
gives dir_ok=False when passed a file, not a directory, then I get a permissions error from:
https://github.com/spack/spack/blob/76ad001c563bfdd1821b590110cbe83e1b177d03/lib/spack/spack/cmd/config.py#L266
when I do `spack config update config` which was requested by `spack` when I tried a `spack install` of Trilinos.

**Comment by cwsmith** [comment]
In Spack v0.18.1 making the change suggested above:

```
$ git diff /opt/scorec/spack/spack0.18.1/lib/spack/spack/cmd/config.py                                                                                                                                                                                                                                                                                                                                        
diff --git a/lib/spack/spack/cmd/config.py b/lib/spack/spack/cmd/config.py
index c6eca2f..bfce9ec 100644
--- a/lib/spack/spack/cmd/config.py
+++ b/lib/spack/spack/cmd/config.py
@@ -259,7 +259,7 @@ def config_update(args):
         cfg_file = spack.config.config.get_config_filename(
             scope.name, args.section
         )
-        scope_dir = scope.path
+        scope_dir = os.path.dirname(scope.path)
         can_be_updated = _can_update_config_file(scope_dir, cfg_file)
         if not can_be_updated:
             if scope.name == 'system':
```

resolved my issue described below.

-------

I’m trying to run spack config update modules within a newly created environment with Spack v0.18.1 using an spack.yaml file I previously used with Spack v0.15.4 and hitting a file permission issue that I can’t sort out.  From what I can tell my user has full access to the files Spack is reporting on. Any thoughts?

```
==> [2022-10-18-12:44:25.261961] Error: Detected permission issues with the following scopes:

        [scope=env:/opt/scorec/spack/spack0.18.1/v0181_0:/opt/scorec/spack/spack0.18.1/v0181_0/modules.yaml, cfg=/opt/scorec/spack/spack0.18.1/v0181_0/modules.yaml]

Either ensure that you have sufficient permissions to modify these files or do not include these scopes in the update.
```

I have write access to the file:

```
$ ls -altr /opt/scorec/spack/spack0.18.1/v0181_0/modules.yaml
-rwxrwxr-x 1 cwsmith softadmin 2902 Oct 18 12:42 /opt/scorec/spack/spack0.18.1/v0181_0/modules.yaml
```

Spack environment settings:

```
$ env | grep SPACK
SPACK_ENV=/opt/scorec/spack/spack0.18.1/v0181_0
SPACK_ROOT=/opt/scorec/spack/spack0.18.1
SPACK_LD_LIBRARY_PATH=/opt/scorec/spack/spack0.18.1/v0181_0/.spack-env/view/lib64:/opt/scorec/spack/spack0.18.1/v0181_0/.spack-env/view/lib
SPACK_USER_CACHE_PATH=/opt/scorec/spack/spack0.18.1/v0181_0
SPACK_DISABLE_LOCAL_CONFIG=true
SPACK_PYTHON=/usr/bin/python3
 eval "if [ -n \"\${${var}-}\" ]; then export SPACK_$var=\${${var}}; fi";
 ```
 
 The full output from `spack -d config update modules` is attached.
[log.gz](https://github.com/spack/spack/files/9813647/log.gz)

**Comment by haampie** [comment]
@cwsmith care to make a PR?

**Comment by cwsmith** [comment]
@haampie Will do.

**Comment by wdconinc** [comment]
Is this still a relevant issue? I am trying to reproduce it so we can maybe get to merging the fix at #33395 (which will need a rebase).

I'm using the following reproducer, which I think should demonstrate this same issue (`install_tree: /tmp` will be updated to `install_tree: root: /tmp`):
```yaml
spack:
  config:
    install_tree: /tmp
  specs:
  - zlib
```

### v0.18.0
```console
docker run --rm -it spack/ubuntu-jammy:v0.18.0
cat << EOF > spack.yaml
spack:
  config:
    install_tree: /tmp
  specs:
  - zlib
EOF
spack env activate -d . 
spack config update config
```
This indeed returns
```console
==> Error: Detected permission issues with the following scopes:

        [scope=env:/root, cfg=/root/spack.yaml]

Either ensure that you have sufficient permissions to modify these files or do not include these scopes in the update.
```

### v0.22.0
(Note container tag does not include `v` anymore.)
```console
docker run --rm -it spack/ubuntu-jammy:0.22.0
cat << EOF > spack.yaml
spack:
  config:
    install_tree: /tmp
  specs:
  - zlib
EOF
spack env activate -d . 
spack config update config
```
which returns
```console
==> The following configuration files are going to be updated to the latest schema format:

        [scope=env:/root, file=/root/spack.yaml]

If the configuration files are updated, versions of Spack that are older than this version may not be able to read them. Spack stores backups of the updated files which can be retrieved with "spack config revert"
==> Do you want to proceed? [y/N] y
==> File "/root/spack.yaml" update [backup=/root/spack.yaml.bkp]
```

So, in summary, I think this issue may have been fixed at some point, and can be closed. Please confirm.
